compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -950,12 +950,12 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     /// suppressed. Needless to say, this must only be set with great care! Cannot be nested.
     pub(super) fn run_for_validation<R>(&self, f: impl FnOnce() -> R) -> R {
         assert!(
-            self.memory.validation_in_progress.replace(true) == false,
+            !self.memory.validation_in_progress.replace(true),
             "`validation_in_progress` was already set"
         );
         let res = f();
         assert!(
-            self.memory.validation_in_progress.replace(false) == true,
```

Comment:
reverted

Reply:
Is this some clippy lint or so? We should probably `allow` it or add a comment or so, otherwise someone else will inevitably try to do the same again.

Reply:
I think it was `needless_bool` but so far I have been hesitant on sprinkling `alllow` attributes all over the codebase.

Reply:
In this case I think it would make a lot of sense. When I wrote the code I already half-expected some clippy-driven cleanup to happen, I just didn't know how to best defend against that.